pages/studentPortal/quizcomponent/FirstProgrammingLanguageQuiz.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import LangResults from "../../../components/resources/quizzes/langQuiz/LangResults";
+import BluePrint from "../../../components/resources/quizzes/shared/BluePrint";
+import SkillSelections from "../../../components/resources/quizzes/shared/SkillSelections";
+import StartQuiz from "../../../components/resources/quizzes/shared/StartQuiz";
+export enum Stage {
+  START,
+  LEARNCODING,
+  WORKFIELDS,
+  BUILDTARGET,
+  BLUEPRINT,
+  RESULTS,
+}
+
+type QuizSelection = {
+  name: string;
+  result?: string;
+  weight?: number;
+};
+
+const FirstProgrammingLanguageQuiz = () => {
+  const quiz: QuizSelection[][] = [
```

Comment:
This data structure is pretty confusing. Right now I read this as an array of an array of QuizSelection objects. 

Each element in the outer array is an array, but each inner array has only one element. So it's kind of pointless.

I think what you're going for is an array of objects with type QuizSelection, which makes sense to me. But you'll need to change your setter functions to reflect this if you change the data structure

Reply:
Good to know. I will have a conversation with Lucky to see next steps!